src/core/structures/PDFAcroForm.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  PDFArray,
+  PDFBool,
+  PDFContext,
+  PDFDict,
+  PDFName,
+  PDFNumber,
+  PDFObject,
+  PDFStream,
+  PDFString,
+} from 'src/core';
+
+class PDFAcroForm extends PDFDict {
+  static fromMapWithContext(
+    dict: Map<PDFName, PDFObject>,
+    context: PDFContext,
+  ): PDFAcroForm {
+    dict.set(
+      PDFName.of('Fields'),
+      dict.get(PDFName.of('Fields')) || PDFArray.withContext(context),
+    );
+    return new PDFAcroForm(dict, context);
+  }
+
+  protected constructor(map: Map<PDFName, PDFObject>, context: PDFContext) {
+    super(map, context);
+  }
+
+  get Fields(): PDFArray {
+    const fieldRefs = this.get(PDFName.of('Fields')) as PDFArray;
+    return fieldRefs.map((fieldRef) => this.context.lookup(fieldRef, PDFDict));
+  }
+
+  get NeedsAppearances(): PDFBool | undefined {
+    return this.get(PDFName.of('NeedsAppearances')) as PDFBool;
+  }

```

Comment:
I can certainly see the appeal of using getter methods here. However, for the sake of API consistency I would prefer that they be normal methods (see [here](https://github.com/Hopding/pdf-lib/blob/620747a044c2b3864c9d844b7c67b924b4f94ea5/src/core/structures/PDFCatalog.ts#L21), [here](https://github.com/Hopding/pdf-lib/blob/620747a044c2b3864c9d844b7c67b924b4f94ea5/src/core/structures/PDFPageTree.ts#L28), and [here](https://github.com/Hopding/pdf-lib/blob/620747a044c2b3864c9d844b7c67b924b4f94ea5/src/core/structures/PDFPageLeaf.ts#L71)).

Also, these seem like perfect opportunities to use typed lookups instead of static casts. For example:
```js
return this.lookupMaybe(PDFName.of('NeedsAppearances'), PDFBool);
```

Reply:
@Hopding  I suppose I don't fully understand the relationship between the contents of a `PDFDict` and its corresponding `PDFContext`. Is the presumption that every object that might be a value in a `PDFDict` has first been registered with the `PDFContext`?

Reply:
> @Hopding I suppose I don't fully understand the relationship between the contents of a `PDFDict` and its corresponding `PDFContext`. Is the assumption that every value in a `PDFDict` has first been registered with the `PDFContext`? I'm not sure whether all objects need to be registered with a context or only indirect objects. If it's only indirect objects, the PDF spec indicates that these dictionary entries are directly accessible, so I think we could just use `get`.

Ah, nvm. I see how this works now. I'll make this change.